packages/fuels-macros/src/setup_program_test/code_gen.rs

```diff
@@ -123,8 +121,13 @@ fn contract_deploying_code(
             let bin_path = project.bin_path();
 
             quote! {
+                // Generate random salt for contract deployment.
+                // These lines must be inside the `quote!` macro, otherwise the salt remains
+                // identical between macro compilation, causing contract id collision.
+                let mut rng = StdRng::from_entropy();
+                let salt: [u8; 32] = rng.gen();
```

Comment:
I did not find a nice place to add `use rand::{rngs::StdRng, Rng, SeedableRng};` so instead I propose we use fully qualified syntax.
```suggestion
                // Generate random salt for contract deployment.
                let mut rng = <rand::rngs::StdRng as rand::SeedableRng>::from_entropy();
                let salt: [u8; 32] = <rand::rngs::StdRng as rand::Rng>::gen(&mut rng);

```